developer/src/kmc-convert/src/keylayout-to-kmn/keylayout-to-kmn-converter.ts

```diff
@@ -1,27 +1,990 @@
+                              /*   rule_type */             "C3",
+                              /*   modifier_prev_deadkey*/  this.create_kmn_modifier(b2_prev_deadkeyModifier_arr[n1][n2], isCapsused),
+                              /*   prev_deadkey */          this.map_UkeleleKC_To_VK(Number(b2_prev_deadkey_arr[n3][0])),
+                              /*   id_prev_deadkey */        dk_counter_C3++,
+                              /*   unique A */              0,
+
+                              /*   modifier_deadkey */      this.create_kmn_modifier(b4_deadkeyModifier_arr[n4][n5], isCapsused),
+                              /*   deadkey */               this.map_UkeleleKC_To_VK(Number(b4_deadkey_arr[n6][0])),
+                              /*   dk for C2*/              0,
+                              /*   unique B */              0,
+
+                              /*   modifier_key*/           b1_modifierKey_arr[n7][3],
+                              /*   key */                   b1_modifierKey_arr[n7][0],
+                              /*   output */                new TextEncoder().encode(b1_modifierKey_arr[n7][4]),
+                              );
+
+                              if ((b1_modifierKey_arr[n7][4] !== undefined)
+                                && (b1_modifierKey_arr[n7][4] !== "undefined")
+                                && (b1_modifierKey_arr[n7][4] !== "")) {
+                                object_array.push(rule_obj);
+                              }
+                            }
+                          }
+                        }
+                      }
+                    }
+                  }
+                }
+              }
+            }
+          }
+        } else {
+
+          /* QUESTION
+          // _S2 why does this not work here? Correct data in this.callback.messages is available !?!?!
+           this.callbacks.reportMessage(ConverterMessages.Error_UnsupportedCharactersDetected({
+               inputFilename: jsonObj.keyboard['@_name'] + ".keylayout",
+               keymap_index: jsonObj.keyboard.keyMapSet[0].keyMap[i]['@_index'],
+               output: jsonObj.keyboard.keyMapSet[0].keyMap[i].key[j]['@_output'],
+               key: jsonObj.keyboard.keyMapSet[0].keyMap[i].key[j]['@_code']
+             }));
+           return null;*/
+
+          console.log("ERROR : some output characters can not be used in Keyman \"",
+            (jsonObj.keyboard['@_name'] + ".keylayout\""),
+            "\"<keyMap index=\"", jsonObj.keyboard.keyMapSet[0].keyMap[i]['@_index'], "\">\" :",
+            jsonObj.keyboard.keyMapSet[0].keyMap[i].key[j]);
+          continue;
+        }
+      }
+    }
+    data_ukelele.arrayOf_Rules = object_array;
+    return this.reviewRuleInputData(data_ukelele);
+  }
+
+  /**
+    * @brief  member function to review data in array of Rules[] of data_ukelele: remove duplicate rules and mark first occurance of a rule in object_array
```

Comment:
I suppose this is a decent reason to not convert on the store level - if there's a legitimate concern for duplicate rules, which could happen if one is store-based and another is defined using only specific values.

... gonna gloss over this method for now, as it seems to require significant .keylayout knowledge, and the examples from the doc referenced earlier aren't explaining enough for me to follow this.